packages/one-app-dev-bundler/esbuild/plugins/externals-loader.js

```
@@ -75,73 +56,35 @@ const externalsLoader = ({ bundleType }) => ({
-})) {
-  try {
-    module.exports = rootModuleExternal.module;
-  } catch (error) {
-    const errorGettingExternal = new Error('Failed to get external ${externalName} from root module on the server');
-    errorGettingExternal.shouldBlockModuleReload = false;
-    throw errorGettingExternal;
-  }
-} else {
-  ${cjsCode}
-}`,
-        };
-      }
       const version = readPackageUpSync({
         cwd: path.resolve(process.cwd(), 'node_modules', externalName),
       })?.packageJson.version;
 
-      if (!version) {
-        throw new Error(`'version' not found for external "${externalName}"`);
-      }
-
-      // client
       return {
         loader: 'js',
-        contents: `\
-try {
-  const fallbackExternal = ${globalReferenceString}.Holocron.getExternal({
-    name: '${externalName}',
-    version: '${version}'
-  });
-  const rootModuleExternal = ${globalReferenceString}.getTenantRootModule && ${globalReferenceString}.getTenantRootModule().appConfig.providedExternals['${externalName}'];
-
-  module.exports = fallbackExternal || (rootModuleExternal ? rootModuleExternal.module : () => {
-    throw new Error('External not found: ${externalName}')
-});
-} catch (error) {
-  const errorGettingExternal = new Error('Failed to get external fallback ${externalName}');
-  errorGettingExternal.shouldBlockModuleReload = false;
-  throw error;
-}`,
+        contents: `
+          try {
+            const Holocron = ${bundleType === BUNDLE_TYPES.SERVER ? 'require("holocron")' : `${globalReferenceString}.Holocron`};
+            const fallbackExternal = Holocron.getExternal({
+              name: '${externalName}',
+              version: '${version}'
+            });
+            const rootModuleExternal = ${globalReferenceString}.getTenantRootModule && ${globalReferenceString}.getTenantRootModule().appConfig.providedExternals['${externalName}'];
+            
+            module.exports = fallbackExternal || (rootModuleExternal ? rootModuleExternal.module : () => {
```

instead of returning a function that throws an error, we should just throw an error here (which would get caught by the below catch)